pkg/trace/api/responses.go

```diff
@@ -46,6 +47,17 @@ func httpDecodingError(err error, tags []string, w http.ResponseWriter) {
 		errtag := "payload-too-large"
 		msg = errtag
 	}
+	// is it a timeout?
+	if err, ok := err.(net.Error); ok && err.Timeout() {
+		status = http.StatusRequestTimeout
+		errtag := "timeout"
+		msg = errtag
+	}
+	// is there something wrong with the content length?
+	if err == io.EOF || err == io.ErrUnexpectedEOF {
+		errtag = "unexpected-eof"
```

Comment:
One last thing: since we're treating both EOF and UnexpectedEOF as the same, can we shorten this to just `eof`?

Reply:
Yes perhaps